### PR TITLE
pkp/pkp-lib#5885 Review remainder update

### PR DESCRIPTION
**Pull request by touhidurabir**
for pkp/pkp-lib#5885

### CODE REVIEW

File: controllers/grid/users/reviewer/form/ReviewerForm.php [hunk 1]
```diff
@@ -73,6 +73,17 @@ public function __construct($submission, $reviewRound)
         $this->addCheck(new \PKP\form\validation\FormValidator($this, 'responseDueDate', 'required', 'editor.review.errorAddingReviewer'));
         $this->addCheck(new \PKP\form\validation\FormValidator($this, 'reviewDueDate', 'required', 'editor.review.errorAddingReviewer'));
 
+        $this->addCheck(
+            new \PKP\form\validation\FormValidatorDateCompare(
+                $this,
+                'reviewDueDate',
+                \Carbon\Carbon::parse(Application::get()->getRequest()->getUserVar('responseDueDate')),
```
Comment (Vitaliy-1) [inline controllers/grid/users/reviewer/form/ReviewerForm.php:80]: May we use `$this->getData('responseDueDate')` here?

File: jobs/email/ReviewRemainder.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        $reviewAssignment = Repo::reviewAssignment()->get($this->reviewAssignmentId);
+        $reviewer = Repo::user()->get($reviewAssignment->getReviewerId());
+        
+        if (!isset($reviewer)) {
+            return;
+        }
+
+        $submission = Repo::submission()->get($this->submissionId);
+        
+        $contextService = Services::get("context");
```
Comment (Vitaliy-1) [inline jobs/email/ReviewRemainder.php:62]: "context" => 'context'

File: jobs/email/ReviewRemainder.php [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+use PKP\log\event\PKPSubmissionEventLogEntry;
+use PKP\core\PKPApplication;
+use PKP\core\Core;
+use PKP\invitation\invitations\ReviewerAccessInvite;
+use PKP\mail\mailables\ReviewResponseRemindAuto;
+use PKP\mail\mailables\ReviewRemindAuto;
+use PKP\jobs\BaseJob;
+
+
+class ReviewRemainder extends BaseJob
```
Comment (Vitaliy-1) [inline jobs/email/ReviewRemainder.php:31]: ReviewReminder

**Review by Vitaliy-1 — CHANGES_REQUESTED** [review]
Thanks @touhidurabir! I left a couple of minor comments but what I'm thinking about is the possibility for journals to set the number of reminders they want rather than predefined before... and after... fields. Have you considered this option?

**Comment by Vitaliy-1** [comment]
@touhidurabir, thinking about this further, one way to achieve this is adding a context setting that depicts a number of email reminders to show. Something like:
```php
$this->addField(new FieldText('numReviewRequestResponseReminders', [
    'label' => __('manager.setup.reviewOptions.numReviewRequestResponseReminders'),
     'description' => __('manager.setup.reviewOptions.numReviewRequestResponseReminders.description'),
     'value' => $context->getData('numReviewRequestResponseReminders'),
     'size' => 'small',
]));
```

And a setting to hold values for all custom fields. Each reminder option should accept negative numbers if implementing this way
```php
foreach ($context->getData('reviewRequestResponseReminders') as $reviewRequestResponseReminder) {
    $this->addField(new FieldText('requestResponseReminder', [
    ...
    ]));
}
```

This also requires Vue form modification to support new fields added/subtracted dynamically - by listening to the `change` event, which is emitted when a field prop changes.

Another way could be adding a new form field component that dynamically supports custom number of inputs depending on a value of another field...

File: classes/components/forms/context/PKPReviewSetupForm.php [hunk 4]
```diff
@@ -92,16 +92,46 @@ public function __construct($action, $locales, $context)
...
 
         if (Config::getVar('general', 'scheduled_tasks')) {
-            $this->addField(new FieldText('numDaysBeforeInviteReminder', [
-                'label' => __('manager.setup.reviewOptions.reminders.response'),
-                'description' => __('manager.setup.reviewOptions.reminders.response.description'),
-                'value' => $context->getData('numDaysBeforeInviteReminder'),
-                'size' => 'small',
-            ]))
-                ->addField(new FieldText('numDaysBeforeSubmitReminder', [
+            // $this->addField(new FieldText('numDaysBeforeInviteReminder', [
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/components/forms/context/PKPReviewSetupForm.php]: removed .

File: classes/components/forms/context/PKPReviewSetupForm.php [hunk 5]
```diff
@@ -92,16 +92,46 @@ public function __construct($action, $locales, $context)
...
+            // ]))
+            //     ->addField(new FieldText('numDaysBeforeSubmitReminder', [
+            //         'label' => __('manager.setup.reviewOptions.reminders.submit'),
+            //         'description' => __('manager.setup.reviewOptions.reminders.submit.description'),
+            //         'value' => $context->getData('numDaysBeforeSubmitReminder'),
+            //         'size' => 'small',
+            //     ]));
+            
+            $this
+                ->addField(new FieldHTML('reviewRequestResponseRemainder', [
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/components/forms/context/PKPReviewSetupForm.php]: fixed.

File: classes/task/ReviewReminder.php [hunk 6]
```diff
@@ -130,35 +61,102 @@ public function executeActions()
...
+
+                if ($reviewAssignment->getDateConfirmed() === null) {
+                    // review request has not been responded
+                    // previous remainder was a BEFORE REVIEW REQUEST RESPONSE remainder
+
+                    if ($dateReminded->lt($dateResponseDue) &&
+                        $currentDate->gte($dateResponseDue) && 
+                        $currentDate->diffInDays($dateResponseDue) >= $numDaysAfterReviewResponseReminderDue) {
+
+                        // ACTION:-> we need to sent a AFTER REVIEW REQUEST RESPONSE remainder
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/task/ReviewReminder.php]: fixed.

File: classes/task/ReviewReminder.php [hunk 7]
```diff
@@ -130,35 +61,102 @@ public function executeActions()
...
+            $dateDue = Carbon::parse($reviewAssignment->getDateDue())->startOfDay();
+
+            if ($reviewAssignment->getDateReminded() !== null) {
+                // we have a remainder sent previously
+
+                $dateReminded = Carbon::parse($reviewAssignment->getDateReminded())->startOfDay();
+
+                if ($reviewAssignment->getDateConfirmed() === null) {
+                    // review request has not been responded
+                    // previous remainder was a BEFORE REVIEW REQUEST RESPONSE remainder
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/task/ReviewReminder.php]: fixed.

File: classes/task/ReviewReminder.php [hunk 8]
```diff
@@ -130,35 +61,102 @@ public function executeActions()
...
+                // the review request has been responded
+                // as long review request has respnded, only need to concern with BEFORE/AFTER REVIEW SUBMIT remainder
+                if ($numDaysAfterReviewSubmitReminderDue && 
+                    $currentDate->gt($dateDue) && 
+                    $currentDate->diffInDays($dateDue) >= $numDaysAfterReviewSubmitReminderDue) {
+                    
+                    // ACTION:-> we need to send AFTER REVIEW SUBMIT remainder
+                    $mailable = ReviewRemindAuto::class;
+
+                } else if ( $numDaysBeforeReviewSubmitReminderDue &&
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/task/ReviewReminder.php:134]: did it to make all the of checks align properly .

File: classes/task/ReviewReminder.php [hunk 9]
```diff
@@ -130,35 +61,102 @@ public function executeActions()
...
+
+                $dateReminded = Carbon::parse($reviewAssignment->getDateReminded())->startOfDay();
+
+                if ($reviewAssignment->getDateConfirmed() === null) {
+                    // review request has not been responded
+                    // previous remainder was a BEFORE REVIEW REQUEST RESPONSE remainder
+
+                    if ($dateReminded->lt($dateResponseDue) &&
+                        $currentDate->gte($dateResponseDue) && 
+                        $currentDate->diffInDays($dateResponseDue) >= $numDaysAfterReviewResponseReminderDue) {
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/task/ReviewReminder.php:98]: fixed.

File: classes/validation/ValidatorDateConparison.php [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+ */
+
+namespace PKP\validation;
+
+use Carbon\Carbon;
+use DateTimeInterface;
+use Exception;
+use PKP\validation\ValidatorFactory;
+
+class ValidatorDateConparison extends Validator
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/validation/ValidatorDateConparison.php]: fixed.

File: classes/form/validation/FormValidatorDateCompare.php [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+     * Constructor.
+     *
+     * @param \PKP\form\Form            $form           the associated form
+     * @param string                    $field          the name of the associated field
+     * @param DateTimeInterface|Carbon  $comparingDate  the comparing date
+     * @param string                    $comparingRule  the comparing rule e.g. equal/greater/lesser/...
+     * @param string                    $type           the type of check, either "required" or "optional"
+     * @param string                    $message        the error message for validation failures (i18n key)
+     */
+    public function __construct(&$form, $field, $comparingDate, $comparingRule,  $type = 'optional', $message = 'email.invalid')
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/form/validation/FormValidatorDateCompare.php]: fixed.

File: controllers/grid/users/reviewer/form/ReviewerForm.php (hunk 1 above)
Comment (touhidurabir) [reply controllers/grid/users/reviewer/form/ReviewerForm.php]: fixed.

File: jobs/email/ReviewRemainder.php (hunk 2 above)
Comment (touhidurabir) [reply jobs/email/ReviewRemainder.php]: fixed.

File: jobs/email/ReviewRemainder.php (hunk 3 above)
Comment (touhidurabir) [reply jobs/email/ReviewRemainder.php]: fixed.

**Comment by touhidurabir** [comment]
Hi @Vitaliy-1 , can you explain how the negative number works in bit more details , I am still bit confused how it will work . I have followed the following implementation approach 

1. set reminder day `before` and `after` the `review request response` due date if invited reviewer has not yet responded to the review invitation request.
2. set reminder day `before` and `after` the `review submission` due date if the reviewer has accepted the invitation but has not submitted the review yet .

The implementation that I have designed is so that it will only sent one such reminder for each case that is 

1. sent only one `review request response` due reminder `before` the response due date if has not yet responded to the request and only if that day value is set .
2. sent only one `review request response` due reminder `after` the response due date if has not yet responded to the request and only if that day value is set .
3. sent only one `review submission` due reminder `before` the review submission due date if reviewer has not yet submitted the review and only if that day value is set .
4. sent only one `review submission` due reminder `after` the review submission due date if reviewer has not yet submitted the review and only if that day value is set .

so for each above 4 cases, only one reminder for each case and no more than that .

**Comment by asmecher** [comment]
Hey @touhidurabir -- please hold off on this one until @Devika008 has had a chance to review things.

File: classes/form/validation/FormValidatorDateCompare.php [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    public function __construct(&$form, $field, $comparingDate, $comparingRule,  $type = 'optional', $message = 'validator.date.comparison')
+    {
+        $validator = new ValidatorDateComparison($comparingDate, $comparingRule);
+        parent::__construct($form, $field, $type, $message, $validator);
+    }
+}
+
+if (!PKP_STRICT_MODE) {
+    class_alias('\PKP\form\validation\FormValidatorDateCompare', '\FormValidatorDateCompare');
```
Comment (asmecher) [inline classes/form/validation/FormValidatorDateCompare.php:45]: These should only be necessary for old code; for new code, just make sure to use FQCNs and `use` statements where they're needed.

File: classes/jats/exceptions/UnableToCreateJATSContentException.php [hunk 13]
```diff
@@ -14,12 +14,17 @@
...
+use Throwable;
 
 class UnableToCreateJATSContentException extends Exception
 {
 
-    public function __construct(public ?Exception $innerException = null)
+    public function __construct(public ?Throwable $innerException = null)
     {
-        parent::__construct(__('publication.jats.defaultContentCreationError'), null, $innerException);
+        parent::__construct(
```
Comment (asmecher) [inline classes/jats/exceptions/UnableToCreateJATSContentException.php:24]: Should these changes be in a different PR?

File: classes/migration/upgrade/v3_5_0/I5885_RenameReviewReminderSettingsName.php [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+{
+    abstract protected function getContextSettingsTable(): string;
+
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        DB::table($this->getContextSettingsTable())
+            ->select(['setting_name'])
```
Comment (asmecher) [inline classes/migration/upgrade/v3_5_0/I5885_RenameReviewReminderSettingsName.php:30]: I think `->select` is dead code here (and below).

File: classes/task/ReviewReminder.php [hunk 15]
```diff
@@ -111,15 +44,13 @@ public function executeActions()
         $context = null;
 
         $contextDao = Application::getContextDAO();
+        $incompleteAssignments = Repo::reviewAssignment()
```
Comment (asmecher) [inline classes/task/ReviewReminder.php:47]: Because you're attempting to keep the `$context` object when possible while iterating through the loop, this would be significantly faster if ordered by context ID. I see that there aren't sorting functions in the review assignment collector, but those should be pretty quick to introduce.

File: classes/task/ReviewReminder.php [hunk 16]
```diff
@@ -130,35 +61,108 @@ public function executeActions()
...
+                // we have a reminder sent previously
+
+                $dateReminded = Carbon::parse($reviewAssignment->getDateReminded())->startOfDay();
+
+                if ($reviewAssignment->getDateConfirmed() === null) {
+                    // review request has not been responded
+                    // previous reminder was a BEFORE REVIEW REQUEST RESPONSE reminder
+
+                    if ($numDaysAfterReviewResponseReminderDue &&
+                        $dateReminded->lt($dateResponseDue) &&
```
Comment (asmecher) [inline classes/task/ReviewReminder.php:96]: I might not be reading this right, but I think this is a misunderstanding of the `date_reminded` column. I believe that gets set back to `null` when a review is confirmed -- see https://github.com/pkp/pkp-lib/blob/20b3178c9f1fb00724a31d73b98eff42131e9ca7/classes/submission/reviewer/ReviewerAction.php#L92..L97.

File: jobs/email/ReviewReminder.php [hunk 17]
```diff
@@ -0,0 +1,117 @@
...
+
+
+class ReviewReminder extends BaseJob
+{
+    protected int $reviewAssignmentId;
+    protected int $submissionId;
+    protected int $contextId;
+    protected string $mailableClass;
+
+    public function __construct(int $reviewAssignmentId, int $submissionId, int $contextId, string $mailableClass)
```
Comment (asmecher) [inline jobs/email/ReviewReminder.php:38]: (Since this is going into the `main` branch, feel free to use [constructor property promotion](https://php.watch/versions/8.0/constructor-property-promotion) if you like.)

**Review by asmecher — CHANGES_REQUESTED** [review]
This is promising, @touhidurabir; I've added some comments, and it also needs some work per Devika's design work to make it friendlier.

File: classes/jats/exceptions/UnableToCreateJATSContentException.php (hunk 13 above)
Comment (touhidurabir) [reply classes/jats/exceptions/UnableToCreateJATSContentException.php:24]: > Should these changes be in a different PR?

that is true but because of this I was having some issue with some of workflow page loading to run some tests

Comment (asmecher) [reply classes/jats/exceptions/UnableToCreateJATSContentException.php:24]: It's already been resolved in `main` in 8ded2f38ebb1572bf0cf47568c51c18684de4a58.

File: classes/task/ReviewReminder.php (hunk 8 above)
[the remark this answers is not in the document]
Comment (asmecher) [reply classes/task/ReviewReminder.php:134]: @touhidurabir, can you check to make sure your pre-commit hooks are working properly? There should be a pre-commit hook to apply a PHP code formatter, so this stuff should be standardized automatically.

File: classes/jats/exceptions/UnableToCreateJATSContentException.php (hunk 13 above)
Comment (touhidurabir) [reply classes/jats/exceptions/UnableToCreateJATSContentException.php:24]: Then it will be removed at next rebase

File: classes/task/ReviewReminder.php (hunk 15 above)
Comment (touhidurabir) [reply classes/task/ReviewReminder.php:47]: In that case , we should also sort by submission id as I am trying to keep the `submission` object also if possible . I added that also .

**Comment by touhidurabir** [comment]
@asmecher take a look at the PR again .

File: classes/components/forms/context/PKPReviewSetupForm.php [hunk 18]
```diff
@@ -45,23 +53,40 @@ public function __construct($action, $locales, $context)
...
-            'options' => [
-                ['value' => ReviewAssignment::SUBMISSION_REVIEW_METHOD_DOUBLEANONYMOUS, 'label' => __('editor.submissionReview.doubleAnonymous')],
-                ['value' => ReviewAssignment::SUBMISSION_REVIEW_METHOD_ANONYMOUS, 'label' => __('editor.submissionReview.anonymous')],
-                ['value' => ReviewAssignment::SUBMISSION_REVIEW_METHOD_OPEN, 'label' => __('editor.submissionReview.open')],
-            ],
-        ]))
+        $this
+            ->addDefaultFields($context)
+            ->addReminderFields($context)
+            ->addReminderDisbaleNoticeField($context);
```
Comment (asmecher) [inline classes/components/forms/context/PKPReviewSetupForm.php:59]: *Disable

File: classes/form/validation/FormValidatorDateCompare.php [hunk 19]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @file classes/form/validation/FormValidatorDateCompare.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
```
Comment (asmecher) [inline classes/form/validation/FormValidatorDateCompare.php:6]: 2024 for new code.

File: classes/jats/exceptions/UnableToCreateJATSContentException.php (hunk 13 above)
Comment (asmecher) [reply classes/jats/exceptions/UnableToCreateJATSContentException.php:23]: This is still showing up in the PR.

File: classes/validation/ValidatorDateComparison.php [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+        self::DATE_COMPARE_RULE_EQUAL               => 'date_equals',
+        self::DATE_COMPARE_RULE_GREATER             => 'after',
+        self::DATE_COMPARE_RULE_LESSER              => 'before',
+        self::DATE_COMPARE_RULE_GREATER_OR_EQUAL    => 'after_or_equal',
+        self::DATE_COMPARE_RULE_LESSER_OR_EQUAL     => 'before_or_equal',
+    ];
+
+    public function __construct(DateTimeInterface|Carbon $comparingDate, string $comparingRule)
+    {
+        if (!in_array($comparingRule, static::getComparingRules())) {
```
Comment (asmecher) [inline classes/validation/ValidatorDateComparison.php:48]: What do you think about using an Enum for this, instead of runtime validation? We've applied this to a few other places in the code, and so far I like it.

File: classes/validation/ValidatorDateComparison.php [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    protected function getValidationApplicableRule(string $rule): mixed
+    {
+        return $this->validationRulesMapping[$rule];
+    }
+}
+
+if (!PKP_STRICT_MODE) {
+    class_alias('\PKP\validation\ValidatorDateComparison', '\ValidatorDateComparison');
```
Comment (asmecher) [inline classes/validation/ValidatorDateComparison.php:96]: We don't need class aliases for new code -- only for backwards compatibility with old code.

File: jobs/email/ReviewReminder.php [hunk 22]
```diff
@@ -0,0 +1,117 @@
...
+    public function __construct(
+        public int $reviewAssignmentId,
+        public int $submissionId,
+        public int $contextId,
+        public string $mailableClass
+    )
+    {
+        parent::__construct();
+
+        $this->reviewAssignmentId = $reviewAssignmentId;
```
Comment (asmecher) [inline jobs/email/ReviewReminder.php:42]: Since you're using promotion, there's no need to assign these in the constructor.

File: jobs/email/ReviewReminder.php [hunk 23]
```diff
@@ -0,0 +1,117 @@
...
+use PKP\mail\mailables\ReviewResponseRemindAuto;
+use PKP\mail\mailables\ReviewRemindAuto;
+use PKP\jobs\BaseJob;
+
+
+class ReviewReminder extends BaseJob
+{
+    public function __construct(
+        public int $reviewAssignmentId,
+        public int $submissionId,
```
Comment (asmecher) [inline jobs/email/ReviewReminder.php:35]: The `$submissionId` and `$contextId` can be derived from the review assignment; would it be better not to serialize them then have to check for discrepancies? (For example, later on we look at the `$submissionId` to fetch the submission from the database -- but there's no check to ensure it matches the submission ID in the review assignment. It would be better just to use the `reviewAssignmentId`, then get the submission ID etc. from that.)

File: locale/en/editor.po [hunk 24]
```diff
@@ -205,6 +205,9 @@ msgstr "Email to be sent to reviewer"
 msgid "editor.review.importantDates"
 msgstr "Important Dates"
 
+msgid "editor.review.importantDates.notice"
+msgstr "Review due date must be greater or euqal to response due date."
```
Comment (asmecher) [inline locale/en/editor.po:209]: *equal

**Review by asmecher — CHANGES_REQUESTED** [review]
Thanks, @touhidurabir; here are some minor changes to suggest. When this is reviewed, could you follow up by posting some screenshots showing the change on the issue, and tag Devika for a look?

File: classes/jats/exceptions/UnableToCreateJATSContentException.php (hunk 13 above)
Comment (touhidurabir) [reply classes/jats/exceptions/UnableToCreateJATSContentException.php:23]: I kept it because we are passing second param as `null` at [here](https://github.com/pkp/pkp-lib/blob/main/classes/jats/exceptions/UnableToCreateJATSContentException.php#L23) which is not allowed as that should be a `int` as specified [here](https://www.php.net/manual/en/class.exception.php)

File: jobs/email/ReviewReminder.php (hunk 23 above)
Comment (touhidurabir) [reply jobs/email/ReviewReminder.php:35]: I think thats better as we intend to keep the job serialized data as small as possible .

File: classes/validation/ValidatorDateComparison.php (hunk 20 above)
Comment (touhidurabir) [reply classes/validation/ValidatorDateComparison.php]: done refactoring to enum .

File: jobs/email/ReviewReminder.php (hunk 23 above)
Comment (touhidurabir) [reply jobs/email/ReviewReminder.php]: However I think we need to set the standard to passing `contextId` whenever it's possible as our `Services` class bit too hard to mock.

File: tests/PKPTestCase.php [hunk 25]
```diff
@@ -205,8 +205,7 @@ protected function mockRequest($path = 'index/test-page/test-op', $userId = null
         $request->setRouter($router);
 
         // Test user.
-        $session = $request->getSession();
-        $session->setUserId($userId);
+        $request->getSessionGuard()->setUserId($userId);
```
Comment (touhidurabir) [inline tests/PKPTestCase.php:208]: a leftover from pkp/pkp-lib#9566. this and above in the function signature .

File: classes/invitation/invitations/ReviewerAccessInvite.php [hunk 26]
```diff
@@ -23,7 +23,7 @@
 use PKP\invitation\invitations\enums\InvitationStatus;
 use PKP\mail\variables\ReviewAssignmentEmailVariable;
 use PKP\security\Validation;
-use ReviewAssignment;
+use PKP\submission\reviewAssignment\ReviewAssignment;
```
Comment (touhidurabir) [inline classes/invitation/invitations/ReviewerAccessInvite.php:26]: This proper namespace update is required for mocking

**Comment by touhidurabir** [comment]
@asmecher can you take another look specifically at the commit [test added for queue job](https://github.com/pkp/pkp-lib/pull/9612/commits/ccbbc264f0cc43448cd6a0d818de29cd23ae39ec) . I have added some tests and at the same time I had to update in the core `PKPTestCase` .

**Comment by touhidurabir** [comment]
@asmecher rebased and all tests are passing . please take a final look .

File: locale/en/manager.po [hunk 27]
```diff
@@ -1238,21 +1238,37 @@ msgstr "Never Remind"
...
-msgstr "Response Reminder"
+msgid "manager.setup.reviewOptions.reminders"
+msgstr "Set Reminders for Review"
 
-msgid "manager.setup.reviewOptions.reminders.response.description"
+msgid "manager.setup.reviewOptions.reminders.description"
 msgstr ""
-"Send an email reminder if a reviewer has not responded to a review request "
-"this many days after the response due date."
+"Send an email reminder before or after for review request response(if reviewer has not responded "
```
Comment (asmecher) [inline locale/en/manager.po:1246]: Missing space before `(`

File: locale/en/manager.po [hunk 28]
```diff
@@ -1238,21 +1238,37 @@ msgstr "Never Remind"
...
+"to review request yet) or review submission (if reviewer has not submitted review yet)"
 
-msgid "manager.setup.reviewOptions.reminders.submit"
-msgstr "Review Reminder"
+msgid "manager.setup.reviewOptions.reminders.response.before"
+msgstr "Response Request Response - Before Due Date"
```
Comment (asmecher) [inline locale/en/manager.po:1250]: Should be "Review Request Response", not "Response Request Response"

File: locale/en/manager.po [hunk 29]
```diff
@@ -1238,21 +1238,37 @@ msgstr "Never Remind"
...
 
-msgid "manager.setup.reviewOptions.reminders.submit.description"
-msgstr ""
-"Send an email reminder if a reviewer has not submitted a recommendation "
-"within this many days after the review's due date."
+msgid "manager.setup.reviewOptions.reminders.response.after"
+msgstr "Response Request Response - After Due Date"
```
Comment (asmecher) [inline locale/en/manager.po:1253]: Also here: should be "Review Request Response", not "Response Request Response"

File: locale/en/manager.po (hunk 27 above)
Comment (touhidurabir) [reply locale/en/manager.po]: fixed.

File: locale/en/manager.po (hunk 28 above)
Comment (touhidurabir) [reply locale/en/manager.po]: fixed.

File: locale/en/manager.po (hunk 29 above)
Comment (touhidurabir) [reply locale/en/manager.po]: fixed.
